Terminal.Gui/View/Layout/ViewLayout.cs

```diff
@@ -616,12 +615,14 @@ public Thickness GetAdornmentsThickness ()
     ///     Helper to get the X and Y offset of the Bounds from the Frame. This is the sum of the Left and Top properties
     ///     of <see cref="Margin"/>, <see cref="Border"/> and <see cref="Padding"/>.
     /// </summary>
-    public Point GetBoundsOffset ()
+    public virtual Point GetBoundsOffset ()
```

Comment:
This is potentially dangerous to be virtual.

But storing the rectangles is a better solution anyway and makes it less of an issue (maybe none at all).

Reply:
It's better than the 3 virtuals that existed before this PR. 

I'm not sure what you mean by "storing rectangles".

Reply:
The value returned by this method is something that only changes if a value it is dependent on changes. Re-calculating it is wasteful, but not the end of the world, since we're on the stack during most of those operations, though it _is_ a hot code path, so impact is potentially worth it.